src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java

```diff
@@ -271,15 +272,39 @@ public FeedbackSessionsCopyFromModal getCopyFromModal() {
             }
         }
         
-        // Add "No active courses" option if there are no active courses
-        if (result.isEmpty()) {
-            ElementTag blankOption = createOption("No active courses!", "", true);
-            result.add(blankOption);
+        // if courses is not empty but result is empty, means the instructor have no permission to edit
+        // rather than the instructor have no active courses
+        // add "No active courses" option if there are no active courses
+        ElementTag defaultOption;
+        if (result.isEmpty() && !courses.isEmpty()) {
+            defaultOption = createOption("No permission to modify courses!", "", true);
+            result.add(defaultOption);
+        } if (result.isEmpty()) {
+            defaultOption = createOption("No active courses!", "", true);
+            result.add(defaultOption);
         }
         
         return result;
     }
     
+    /**
+     * Checks if the instructor have modify permission for any of the courses that he is in.
+     * @param courses list of courses
+     * @param instructors instructor and the courses that he is in
+     * @return true if he does not have modify permission for all the courses that he is in
+     */
+    private boolean isNoModifyPermissionForAllCourses(List<CourseAttributes> courses, Map<String, InstructorAttributes> instructors) {
+        for (CourseAttributes course : courses) {
+            InstructorAttributes instructor = instructors.get(course.id);
+            boolean isAllowedForPriviledge = instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION);
```

Comment:
Likewise, it's the sessions, not the course.

And seeing as this loop and conditional is already done at L253 - 268, shouldn't it be possible to re-use the result somehow?
Maybe calculate this value closer to the start before using it for both purposes?
I would honestly recommend just extracting the `defaultOption` segment into another method and using it from `buildBasicForm()` or something.